test/e2e/bgppolicy_test.go

```diff
@@ -0,0 +1,319 @@
+		require.Equal(t, 0, rc)
+		require.Contains(t, stdout, "agnhost-")
+	}
+
+	t.Log("Update the BGP configuration on the remote FRR router")
+	configureFRRRouterBGP(t, remoteASN, updatedLocalASN)
+
+	_, err = data.updateServiceInternalTrafficPolicy("agnhost-svc", true)
+	require.NoError(t, err)
+
+	t.Log("Update the test BGPPolicy")
+	updatedBGPPolicy := bgpPolicy.DeepCopy()
+	updatedBGPPolicy.Spec.LocalASN = updatedLocalASN
+	updatedBGPPolicy.Spec.Advertisements.Pod = nil
+	_, err = data.crdClient.CrdV1alpha1().BGPPolicies().Update(context.TODO(), updatedBGPPolicy, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Log("Get routes installed on remote FRR router and verify them")
+	expectedRoutes = []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: []string{nodeIPv4(0)}}}
+	expectedRouteStrings = routesToStrings(expectedRoutes)
+	assert.EventuallyWithT(t, func(tc *assert.CollectT) {
+		gotRoutes := dumpFRRRouterBGPRoutes()
+		gotRouteStrings := routesToStrings(gotRoutes)
+		for _, expectedRouteString := range expectedRouteStrings {
+			assert.Contains(tc, gotRouteStrings, expectedRouteString)
+		}
+	}, 30*time.Second, time.Second)
+
+	t.Log("verify the connectivity of the installed routes on remote FRR route")
+	ipsToConnect = []string{clusterIP}
+	for _, ip := range ipsToConnect {
+		cmd := fmt.Sprintf("/usr/bin/wget -O - http://%s:8080/hostname -T 5", ip)
+		rc, stdout, _, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, cmd, "/", nil, "")
+		require.NoError(t, err)
+		require.Equal(t, 0, rc)
+		require.Contains(t, stdout, "agnhost-")
+	}
+}
+
+func configureFRRRouterBGP(t *testing.T, localASN, remoteASN int32) {
+	frrCommands := []string{
+		"configure terminal",
+		fmt.Sprintf("router bgp %d", localASN),
+		"no bgp ebgp-requires-policy",
+		"no bgp network import-check",
+	}
+	for _, node := range clusterInfo.nodes {
+		frrCommands = append(frrCommands, fmt.Sprintf("neighbor %s remote-as %d", node.ipv4Addr, remoteASN))
+		frrCommands = append(frrCommands, fmt.Sprintf("neighbor %s password %s", node.ipv4Addr, password))
+	}
+	frrCommands = append(frrCommands,
+		"exit",
+		"exit",
+		"write memory")
+
+	rc, stdout, stderr, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, "/usr/bin/vtysh", "/", nil, strings.Join(frrCommands, "\n"))
+	t.Log(stdout)
+	t.Log(stderr)
+	require.NoError(t, err, fmt.Sprintf("error when running FRR commands '%v'", frrCommands))
+	require.Equal(t, 0, rc)
+}
+
+func cleanupFRRRouterBGP(t *testing.T, asn int32) {
+	frrCommands := []string{
+		"configure terminal",
+		fmt.Sprintf("no router bgp %d", asn),
+		"exit",
+		"write memory",
+	}
+
+	rc, stdout, stderr, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, "/usr/bin/vtysh", "/", nil, strings.Join(frrCommands, "\n"))
+	t.Log(stdout)
+	t.Log(stderr)
+	require.NoError(t, err, fmt.Sprintf("error when running FRR commands '%v'", frrCommands))
+	require.Equal(t, 0, rc)
+}
+
+func dumpFRRRouterBGPRoutes() []FRRRoute {
```

Comment:
I think it's not ideal that this function doesn't return an error when applicable, as in case of error the caller will keep looping until the timeout is reached.
I know that it's because of a current limitation of `assert.EventuallyWithT` (at least until the next release of testify - see https://github.com/stretchr/testify/issues/1396).
For now, I would recommend defining an additional helper function, `checkFRRRouterBGPRoutes` which will use `wait.Poll...` to call `dumpFRRRouterBGPRoutes` in a loop. In case of error, `checkFRRRouterBGPRoutes` can fail immediately. The function should also probably call `t.Helper` for improved test debuggability.